### PR TITLE
feat: add linkExpiry to WaitlistedCamper model

### PR DESCRIPTION
**Pull request by NirmalHegde**
## Notion ticket link

[Ticket Name](https://www.notion.so/uwblueprintexecs/Waitlist-Model-Update-Setting-invite-expires-on-field-f5c661ad32d244108fcbbe2f412fa3e9)

## Implementation description
* Added linkExpiry to model

## Steps to test
1. add waitlisted camper using the /campers/waitlist endpoint (POST)
2. grab waitlistedCamper id and use /campers/waitlist/:waitlistedCamperId (PATCH)
3. check if link expiry exists in Atlas and is 3 days ahead

## What should reviewers focus on?
* Ensure time is 3 days ahead

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [ ] I have run the appropriate linter(s)
- [ ] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: backend/typescript/services/implementations/camperService.ts [hunk 1]
```diff
@@ -803,7 +804,7 @@ class CamperService implements ICamperService {
   }
 
   /* eslint-disable consistent-return */
-  async inviteWaitlistedCamper(waitlistedCamperId: string): Promise<any> {
+  async inviteWaitlistedCamper(waitlistedCamperId: string): Promise<unknown> {
```
Comment (annechung) [inline backend/typescript/services/implementations/camperService.ts:807]: do we have to change this?

**Review by annechung — APPROVED** [review]
looks good!!

File: backend/typescript/services/implementations/camperService.ts [hunk 2]
```diff
@@ -302,6 +302,7 @@ class CamperService implements ICamperService {
         contactEmail: waitlistedCamper.contactEmail,
         contactNumber: waitlistedCamper.contactNumber,
         campSession: waitlistedCamper.campSession,
+        linkExpiry: undefined,
```
Comment (annechung) [inline backend/typescript/services/implementations/camperService.ts:305]: do we need this if linkExpiry is optional

File: backend/typescript/services/implementations/camperService.ts (hunk 1 above)
Comment (NirmalHegde) [reply backend/typescript/services/implementations/camperService.ts:807]: it kept giving warnings and it annoyed me lol so I removed.

File: backend/typescript/services/implementations/camperService.ts (hunk 2 above)
Comment (NirmalHegde) [reply backend/typescript/services/implementations/camperService.ts:305]: I thought it would be nice so they know its a field, but I guess not

**Pull request merged**